transcripts/chaincode-labs/2019-06-12-ethan-heilman-network-partitioning-attacks.mdwn

```diff
@@ -0,0 +1,135 @@
+Ethan Heilman
+
+Bitcoin network partitioning & network-level privacy attacks
+
+Location: Chaincode Labs 2019 Residency
+
+Video: <https://www.youtube.com/watch?v=StnOVBbIpD8>
+
+Slides: <https://residency.chaincode.com/presentations/bitcoin/ethan_heilman_p2p.pdf>
+
+Eclipse attack paper: <https://eprint.iacr.org/2015/263.pdf>
+
+
+# Bitcoin’s P2P Network
+
+Please interrupt me. I'm going to go through this stuff a little bit quickly because I hear that you've already read [Eclipse attack paper](https://eprint.iacr.org/2015/263.pdf), and you've already looked at some of the peer to peer.
+
+Bitcoin uses a peer-to-peer network to announce transactions and blocks it makes 8 outgoing connections. These are TCP connections initiated by the node. It makes up to 116 connections that are incoming, so they're initiated by someone else. And it makes one feeler connection under some circumstances, and I'll talk more about this feeler connection later. You can actually treat the feeler connection as an outgoing connection. It is an outgoing connection, but it has some different rules, so I just broke it out separately, but if you want to say that there are nine outgoing connections counting the one failure connection, you can say that as well. And I want to point out that when we talk about outgoing connections, we're always talking from the perspective of a node, so if you look at this purple node, it has an outgoing connection to this yellowish node. But from the yellowish node's perspective, this is an incoming connection. Whenever we talk about outgoing and incoming, it's really which node is an outgoing one everyone's outgoing connection is another's nodes incoming connection.
```

Comment:
...and blocks. **It makes** 8 outgoing connections. 

....is an outgoing one. **Everyone's** outgoing connection is another node's incoming connection